### PR TITLE
Fix: Stretch Text overflows into the padding area.

### LINKED ISSUE #73218: Stretch Text doesn't work correctly with padding
### Description

The text overflows on the right side. I would expect the text to fit _inside_ the padding.

cc @jorgefilipecosta 

### Step-by-step reproduction instructions

- Insert a Stretch Heading or Paragraph block.
- Apply left or right padding

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/f76a63c5-485b-42d4-83d8-faaec4cb35e5

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Fixes https://github.com/WordPress/gutenberg/issues/73218.
Fixes an issue reported by @t-hamano where if there is padding on the paragraph the text may overflow into the padding area (it fits the container but the padding area stops being a real padding).

## How?
Paddings does not changes scrollWidth and clientWidth, which were using for the binary search, in order to fix the issue we need to get the real text dimensions, and the available space.

To get the real text dimensions of a text node the way to do it is using a range plus getBoundingClientRect (as far as I know that's the only way).

To get the available width we subtract the paddings to the clientWidth.

## Video

https://github.com/user-attachments/assets/51c079e8-90bc-4661-8c54-f276c8319563

(works the same as a normal paragraph)

## Testing Instructions
Add a Stretch Paragraph, add some padding.
Verify no matter the text or padding the text never overflows into the padding area.

### CODE REVIEW

**Review by t-hamano — APPROVED** [review]
LGTM! Works as expected.

File: packages/block-editor/src/utils/fit-text-utils.js [hunk 1]
```diff
@@ -33,6 +50,7 @@ function findOptimalFontSize( textElement, applyFontSize ) {
 			maxSize = midSize - 1;
 		}
 	}
+	range.detach();
```
Comment (mcsf) [inline packages/block-editor/src/utils/fit-text-utils.js:53]: @jorgefilipecosta, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Range/detach#browser_compatibility), `Range#detach` is not Firefox-compatble.

Comment (mcsf) [reply packages/block-editor/src/utils/fit-text-utils.js:53]: More importantly:

> The Range.detach() method does nothing. It used to disable the [Range](dfile:///Users/mcsf/Library/Application%20Support/Dash/DocSets/JavaScript/JavaScript.docset/Contents/Resources/Documents/developer.mozilla.org/en-US/docs/Web/API/Range.html) object and enable the browser to release associated resources. The method has been kept for compatibility.
